manifests/init.pp

```diff
@@ -55,31 +55,29 @@
 # [*accept_eth1*]
 #   Whether or not to accept connections to Docker containers from the eth1
 #   interface.
+#
+# [*accept_rules*]
+#   A hash of firewall resources to create. These rules will apply to the
+#   DOCKER_INPUT chain and jump to the DOCKER chain so the connection is
+#   accepted if it is really headed for a container. All other parameters for
+#   the firewall resource can be set by the user.
 class docker_firewall (
-  $bridges                      = {},
-  $default_bridges              = {'docker0' => {}},
-
-  $prerouting_nat_purge_ignore  = [],
-  $prerouting_nat_policy        = undef,
-  $output_nat_purge_ignore      = [],
-  $output_nat_policy            = undef,
-  $postrouting_nat_purge_ignore = [],
-  $postrouting_nat_policy       = undef,
-  $forward_filter_purge_ignore  = [],
-  $forward_filter_policy        = 'drop',
-
-  $accept_eth0                  = false,
-  $accept_eth1                  = false,
+  Hash[String, Hash] $bridges                      = {},
+  Hash[String, Hash] $default_bridges              = {'docker0' => {}},
+
+  Variant[String, Array[String]] $prerouting_nat_purge_ignore  = [],
+  Optional[String]               $prerouting_nat_policy        = undef,
+  Variant[String, Array[String]] $output_nat_purge_ignore      = [],
+  Optional[String]               $output_nat_policy            = undef,
+  Variant[String, Array[String]] $postrouting_nat_purge_ignore = [],
+  Optional[String]               $postrouting_nat_policy       = undef,
+  Variant[String, Array[String]] $forward_filter_purge_ignore  = [],
+  Optional[String]               $forward_filter_policy        = 'drop',
+
+  Boolean            $accept_eth0  = false,
+  Boolean            $accept_eth1  = false,
```

Comment:
I think let's remove these here, since we're breaking compat by moving to Puppet 4 anyway.